plugins/obs-filters/nvafx-load.h

```diff
@@ -0,0 +1,159 @@
+	/** Model file could not be loaded */
+	NVAFX_STATUS_MODEL_LOAD_FAILED = 8,
+
+	/** (32 bit SDK only) COM server was not registered, please see user manual for details */
+	NVAFX_STATUS_32_SERVER_NOT_REGISTERED = 9,
+	/** (32 bit SDK only) COM operation failed */
+	NVAFX_STATUS_32_COM_ERROR = 10,
+	/** GPU supported. The SDK requires Turing and above GPU with Tensor cores */
+	NVAFX_STATUS_GPU_UNSUPPORTED = 11,
+} NvAFX_Status;
+
+typedef const char *NvAFX_EffectSelector;
+typedef const char *NvAFX_ParameterSelector;
+typedef void *NvAFX_Handle;
+
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_GetEffectList_t)(int *num_effects,
+					   NvAFX_EffectSelector *effects[]);
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_CreateEffect_t)(NvAFX_EffectSelector code,
+					  NvAFX_Handle *effect);
+typedef NvAFX_Status NVAFX_API (*NvAFX_DestroyEffect_t)(NvAFX_Handle effect);
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_SetU32_t)(NvAFX_Handle effect,
+				    NvAFX_ParameterSelector param_name,
+				    unsigned int val);
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_SetString_t)(NvAFX_Handle effect,
+				       NvAFX_ParameterSelector param_name,
+				       const char *val);
+typedef NvAFX_Status NVAFX_API (*NvAFX_SetFloat_t)(
+	NvAFX_Handle effect, NvAFX_ParameterSelector param_name, float val);
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_GetU32_t)(NvAFX_Handle effect,
+				    NvAFX_ParameterSelector param_name,
+				    unsigned int *val);
+typedef NvAFX_Status
+	NVAFX_API (*NvAFX_GetString_t)(NvAFX_Handle effect,
+				       NvAFX_ParameterSelector param_name,
+				       char *val, int max_length);
+typedef NvAFX_Status NVAFX_API (*NvAFX_GetFloat_t)(
+	NvAFX_Handle effect, NvAFX_ParameterSelector param_name, float *val);
+typedef NvAFX_Status NVAFX_API (*NvAFX_Load_t)(NvAFX_Handle effect);
+typedef NvAFX_Status NVAFX_API (*NvAFX_Run_t)(NvAFX_Handle effect,
+					      const float **input,
+					      float **output,
+					      unsigned num_samples,
+					      unsigned num_channels);
+
+static NvAFX_GetEffectList_t NvAFX_GetEffectList = NULL;
+static NvAFX_CreateEffect_t NvAFX_CreateEffect = NULL;
+static NvAFX_DestroyEffect_t NvAFX_DestroyEffect = NULL;
+static NvAFX_SetU32_t NvAFX_SetU32 = NULL;
+static NvAFX_SetString_t NvAFX_SetString = NULL;
+static NvAFX_SetFloat_t NvAFX_SetFloat = NULL;
+static NvAFX_GetU32_t NvAFX_GetU32 = NULL;
+static NvAFX_GetString_t NvAFX_GetString = NULL;
+static NvAFX_GetFloat_t NvAFX_GetFloat = NULL;
+static NvAFX_Load_t NvAFX_Load = NULL;
+static NvAFX_Run_t NvAFX_Run = NULL;
+
+void release_lib(void)
+{
+	NvAFX_GetEffectList = NULL;
+	NvAFX_CreateEffect = NULL;
+	NvAFX_DestroyEffect = NULL;
+	NvAFX_SetU32 = NULL;
+	NvAFX_SetString = NULL;
+	NvAFX_SetFloat = NULL;
+	NvAFX_GetU32 = NULL;
+	NvAFX_GetString = NULL;
+	NvAFX_GetFloat = NULL;
+	NvAFX_Load = NULL;
+	NvAFX_Run = NULL;
+	if (nv_audiofx) {
+		FreeLibrary(nv_audiofx);
+		nv_audiofx = NULL;
+	}
+}
+
+static void nvafx_get_sdk_path(char buffer[MAX_PATH])
```

Comment:
Because this function can fail, this function should ideally return a boolean to determine success or failure.

Reply:
fixed and the return bool is now used to stop the loading in case of error